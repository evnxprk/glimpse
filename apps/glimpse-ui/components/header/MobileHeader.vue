<template>
  <VNavigationDrawer v-model="localValue" app temporary dark>
    <VList nav>
      <!-- Header -->
      <VListItem class="drawer-header">
        <img src="../../assets/rpitv_logo_400.png" alt="Logo" class="tv-logo">
        <h3>RPI TV</h3>
      </VListItem>
      <VListItemSubtitle class="drawer-social">
        <a
          v-for="item in socials"
          :key="item.title"
          :href="item.path"
          :aria-label="item.title"
        >
          <font-awesome-icon :icon="['fab', item.icon]" size="lg" />
        </a>
      </VListItemSubtitle>

      <VDivider />

      <!-- Left menu items -->
      <span v-for="item in items.LEFT">
        <VListItem
          v-if="(!item.sublist || item.sublist.length === 0) && (!item.conditional || item.conditional())"
          :key="item.title"
          :to="item.path"
          :nuxt="true"
        >
          <VListItemIcon>
            <font-awesome-icon :icon="item.icon" class="drawer-icon" />
          </VListItemIcon>

          <VListItemTitle>
            {{ item.title }}
          </VListItemTitle>
        </VListItem>

        <HeaderDropdown
          v-else-if="item.sublist && item.sublist.length > 0 && (!item.conditional || item.conditional())"
          :title="item.title"
          :items="item.sublist"
          :icon="item.icon"
          :depth="1"
        />
      </span>

      <VDivider />

      <!-- Right menu items -->
      <span v-for="item in items.RIGHT">
        <VListItem
          v-if="(!item.sublist || item.sublist.length === 0) && (!item.conditional || item.conditional())"
          :key="item.title"
          :to="item.path"
          :class="item.customClass"
          :nuxt="true"
        >
          <VListItemIcon>
            <font-awesome-icon :icon="item.icon" class="drawer-icon" />
          </VListItemIcon>

          <VListItemTitle>
            {{ item.title }}
          </VListItemTitle>
        </VListItem>

        <HeaderDropdown
          v-else-if="item.sublist && item.sublist.length > 0 && (!item.conditional || item.conditional())"
          :title="item.title"
          :items="item.sublist"
          :icon="item.icon"
          :class="item.customClass"
          :depth="1"
        />
      </span>
    </VList>
  </VNavigationDrawer>
</template>

<script>
import HeaderDropdown from '~/components/header/HeaderDropdown'
export default {
  name: 'MobileHeader',
  components: { HeaderDropdown },
  props: {
    items: {
      type: Object,
      required: true
    },
    value: {
      type: Boolean,
      default: false
    }
  },
  data () {
    return {
      socials: [
        { title: 'YouTube', path: 'https://youtube.com/rpitv', icon: 'youtube' },
        { title: 'Twitter', path: 'https://twitter.com/rpitv', icon: 'twitter' },
        { title: 'Reddit', path: 'https://reddit.com/u/rpi_tv', icon: 'reddit-alien' }
      ]
    }
  },
  computed: {
    localValue: {
      get () {
        return this.value
      },
      set (newVal) {
        this.$emit('input', newVal)
      }
    }
  }
}
</script>

<style lang="scss" scoped>

  /* Drawer */
  .drawer-icon {
    --fa-primary-color: #b05454;
    --fa-secondary-color: #f56e6c;
    --fa-secondary-opacity: 1.0;
  }

  .drawer-header .tv-logo {
    width: 75px;
    margin-right: 10px;
  }
  .drawer-social {
    position: relative;
    margin-bottom: 10px;
    top: -10px;
    left: 10px;

    * {
      margin-right: 10px;
    }
  }
</style>
