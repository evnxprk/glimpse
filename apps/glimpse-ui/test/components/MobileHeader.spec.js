import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'
import VuetifyOptions from '../../vuetify.config'
import MobileHeader from '~/components/header/MobileHeader'

Vue.use(VueRouter)
Vue.use(Vuetify)
const vuetifyInstance = new Vuetify(VuetifyOptions)

describe('MobileHeader', () => {
  let wrapper
  let conditionalA
  let conditionalSubB
  let layoutDesc

  beforeEach(() => {
    conditionalA = jest.fn()
    conditionalSubB = jest.fn()
    layoutDesc = {
      LEFT: [
        {
          'title': 'ElemA',
          'path': '/a',
          'icon': ['fal', 'a'],
          'sublist': [],
          'customClass': 'a-class',
          'conditional': conditionalA
        },
        {
          'title': 'ElemB',
          'path': '/b',
          'icon': ['fal', 'b'],
          'sublist': [
            {
              'title': 'BSubElem',
              'path': '/b/sub',
              'icon': ['fal', 'b-sub'],
              'sublist': [],
              'customClass': '',
              'conditional': conditionalSubB
            }
          ],
          'customClass': 'b-class',
          'conditional': null
        }
      ],
      RIGHT: [
        {
          'title': 'ElemC',
          'path': '/c',
          'icon': ['fal', 'c-sub'],
          'sublist': [],
          'customClass': '',
          'conditional': null
        }
      ]
    }
  })

  function mount (opts) {
    wrapper = shallowMount(MobileHeader, {
      ...opts,
      vuetify: vuetifyInstance,
      router: new VueRouter(),
      stubs: {
        NuxtLink: RouterLinkStub
      } })
  }

  test('Properly renders', () => {
    mount({
      propsData: {
        items: layoutDesc,
        value: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(conditionalA).toBeCalled()
    expect(wrapper.findAll('headerdropdown-stub').length).toEqual(1)
  })
})
