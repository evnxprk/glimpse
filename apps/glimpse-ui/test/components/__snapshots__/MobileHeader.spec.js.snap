// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileHeader Properly renders 1`] = `
<vnavigationdrawer-stub
  app="true"
  dark="true"
  height="100vh"
  minivariantwidth="80"
  mobilebreakpoint="1264"
  src=""
  tag="nav"
  temporary="true"
  value="true"
  width="256"
>
  <vlist-stub
    nav="true"
    tag="div"
    tile="true"
  >
    <vlistitem-stub
      activeclass=""
      class="drawer-header"
      tag="div"
    >
      <img
        alt="Logo"
        class="tv-logo"
        src="../../assets/rpitv_logo_400.png"
      />
       
      <h3>
        RPI TV
      </h3>
    </vlistitem-stub>
     
    <vlistitemsubtitle-stub
      class="drawer-social"
    >
      <a
        aria-label="YouTube"
        href="https://youtube.com/rpitv"
      >
        <font-awesome-icon-stub
          icon="fab,youtube"
          size="lg"
        />
      </a>
      <a
        aria-label="Twitter"
        href="https://twitter.com/rpitv"
      >
        <font-awesome-icon-stub
          icon="fab,twitter"
          size="lg"
        />
      </a>
      <a
        aria-label="Reddit"
        href="https://reddit.com/u/rpi_tv"
      >
        <font-awesome-icon-stub
          icon="fab,reddit-alien"
          size="lg"
        />
      </a>
    </vlistitemsubtitle-stub>
     
    <vdivider-stub />
     
    <span>
      <!---->
    </span>
    <span>
      <headerdropdown-stub
        depth="1"
        icon="fal,b"
        items="[object Object]"
        title="ElemB"
      />
    </span>
     
    <vdivider-stub />
     
    <span>
      <vlistitem-stub
        activeclass=""
        class=""
        nuxt="true"
        tag="div"
        to="/c"
      >
        <vlistitemicon-stub>
          <font-awesome-icon-stub
            class="drawer-icon"
            icon="fal,c-sub"
          />
        </vlistitemicon-stub>
         
        <vlistitemtitle-stub>
          
          ElemC
        
        </vlistitemtitle-stub>
      </vlistitem-stub>
    </span>
  </vlist-stub>
</vnavigationdrawer-stub>
`;
